packages/wallet/src/KeyManagement/LedgerKeyAgent.ts

```diff
@@ -164,6 +206,46 @@ export class LedgerKeyAgent extends KeyAgentBase {
     });
   }
 
+  async signTransaction(
+    { body }: TxInternals,
+    { inputAddressResolver }: SignTransactionOptions
+  ): Promise<Cardano.Signatures> {
+    try {
+      const { accountIndex, networkId, communicationType, knownAddresses } = this
+        .serializableData as SerializableLedgerKeyAgentData;
+      const cslTxBody = coreToCsl.txBody(body);
+      const addressHex = utils.buf_to_hex(utils.bech32_decodeAddress(this.knownAddresses[0].address.toString()));
+      const ledgerTxData = await txToLedger({
+        accountIndex,
+        addressHex,
```

Comment:
This works when used with SingleAddessWallet, but in general it might have multiple addresses. You should probably remove this, so the mapper can check all knownAddresses to find if it's a change output.